cudapoa/src/cudapoa_kernels.cuh

```diff
@@ -419,279 +493,119 @@ void generatePOA(genomeworks::cudapoa::OutputDetails* output_details_d,
     uint16_t* outgoing_edges_coverage_count = graph_details_d->outgoing_edges_coverage_count;
     SizeT* node_id_to_msa_pos               = graph_details_d->node_id_to_msa_pos;
 
-    int32_t nwindows_per_block = CUDAPOA_THREADS_PER_BLOCK / WARP_SIZE;
-    int32_t nblocks            = (total_windows + nwindows_per_block - 1) / nwindows_per_block;
+    int32_t nwindows_per_block         = CUDAPOA_THREADS_PER_BLOCK / WARP_SIZE;
+    int32_t nblocks                    = (cuda_banded_alignment || cuda_adaptive_banding) ? total_windows : (total_windows + nwindows_per_block - 1) / nwindows_per_block;
```

Comment:
We can drop the `||` if banded flag is always on when adaptive is on.

Reply:
In the earlier versions, these two flags were independent, but as you mentioned, currently the're dependent. But I think keeping `||` makes it less prone to possible bugs in future changes, since we do not rely on any implicit assumptions that are subject to change.